### PR TITLE
Rename downward-api files to be consistent

### PR DESCRIPTION
**Pull request by tmrts**
Renames underscored downward API file names for consistency.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/28146)

### CODE REVIEW

**Comment by tmrts** [comment]
@pmorie PTAL

File: docs/design/downwardapi_resources_limits_requests.md [hunk 1]
```diff
@@ -18,11 +18,6 @@
 If you are using a released version of Kubernetes, you should
 refer to the docs that go with that version.
 
-<!-- TAG RELEASE_LINK, added by the munger automatically -->
-<strong>
-The latest release of this document can be found
-[here](http://releases.k8s.io/release-1.3/docs/design/downward_api_resources_limits_requests.md).
-
```
Comment (yifan-gu) [inline docs/design/downwardapi_resources_limits_requests.md]: Why this is removed.

Comment (tmrts) [inline docs/design/downwardapi_resources_limits_requests.md]: Munger script removed it and verify-munge passes as well, not sure what to make of it

Comment (tmrts) [inline docs/design/downwardapi_resources_limits_requests.md]: @eparis can you PTAL at munger behavior for this tiny PR?

Comment (yifan-gu) [inline docs/design/downwardapi_resources_limits_requests.md]: Hmm, seems after changing the name, the script cannot locate the file at https://github.com/kubernetes/kubernetes/blob/release-1.3/docs/design/downward_api_resources_limits_requests.md

cc @david-mcmahon

**Pull request closed without merging**